src/libraries/System.Formats.Tar/tests/TarFile/TarFile.CreateFromDirectory.File.Tests.cs

```diff
@@ -193,5 +193,65 @@ public void IncludeAllSegmentsOfPath(bool includeBaseDirectory)
 
             Assert.Null(reader.GetNextEntry());
         }
+
+        [Fact]
+        public void SkipRecursionIntoDirectorySymlinks()
+        {
+            using TempDirectory root = new TempDirectory();
+
+            string destinationArchive = Path.Join(root.Path, "destination.tar");
+
+            string externalDirectory = Path.Join(root.Path, "externalDirectory");
+            Directory.CreateDirectory(externalDirectory);
+
+            File.Create(Path.Join(externalDirectory, "file.txt")).Dispose();
+
+            string sourceDirectoryName = Path.Join(root.Path, "baseDirectory");
+            Directory.CreateDirectory(sourceDirectoryName);
+
+            string subDirectory = Path.Join(sourceDirectoryName, "subDirectory");
+            Directory.CreateSymbolicLink(subDirectory, externalDirectory); // Should not recurse here
+
+            TarFile.CreateFromDirectory(sourceDirectoryName, destinationArchive, includeBaseDirectory: false);
+
+            using FileStream archiveStream = File.OpenRead(destinationArchive);
+            using (TarReader reader = new(archiveStream, leaveOpen: false))
+            {
+                TarEntry entry = reader.GetNextEntry();
+                Assert.NotNull(entry);
+                Assert.Equal("subDirectory/", entry.Name);
+
+                Assert.Null(reader.GetNextEntry()); // file.txt should not be found
+            }
+        }
+
+        [Fact]
+        public void SkipRecursionIntoBaseDirectorySymlink()
+        {
+            using TempDirectory root = new TempDirectory();
+
+            string destinationArchive = Path.Join(root.Path, "destination.tar");
+
+            string externalDirectory = Path.Join(root.Path, "externalDirectory");
+            Directory.CreateDirectory(externalDirectory);
+
+            string subDirectory = Path.Join(externalDirectory, "subDirectory");
+            Directory.CreateDirectory(subDirectory);
+
+            string sourceDirectoryName = Path.Join(root.Path, "baseDirectory");
+            Directory.CreateSymbolicLink(sourceDirectoryName, externalDirectory);
+
+            TarFile.CreateFromDirectory(sourceDirectoryName, destinationArchive, includeBaseDirectory: true); // Base directory is a symlink, do not recurse
```

Comment:
Is this how `tar` works? Doesn't it resolve the link when it is the base directory?

Reply:
Yes:

```bash
$ cd ~
$ mkdir root
$ cd root
$ mkdir externalDirectory
$ mkdir externalDirectory/baseDirectory
$ ln -s externalDirectory/ baseDirectory
$ ls
baseDirectory externalDirectory
ls externalDirectory
subDirectory
$ ls baseDirectory
subDirectory
$ tar -cvf destination.tar baseDirectory
baseDirectory
```

If I open `destination.tar` with an hex editor, you'll see a single entry, the base directory, without following the symlink:

![image](https://user-images.githubusercontent.com/1175054/186018525-fc64180f-1a4e-4bc2-8905-6f044012562e.png)